engineapi/types.go

```diff
@@ -179,6 +179,9 @@ type Backup struct {
 	VolumeBackingImageName string               `json:"volumeBackingImageName"`
 	Messages               map[string]string    `json:"messages"`
 	CompressionMethod      string               `json:"compressionMethod"`
+	Parameters             map[string]string    `json:"parameters"`
+	NewlyUploadedDataSize  string               `json:"newlyUploadedDataSize"`
+	ReUploadedDataSize     string               `json:"reUploadedDataSize"`
```

Comment:
Why not use `int`?

Reply:
In the backupstore lib
we present is as string in the json
```
type Backup struct {
	sync.Mutex
	Name                  string
	VolumeName            string
	SnapshotName          string
	SnapshotCreatedAt     string
	CreatedTime           string
	Size                  int64 `json:",string"`
	Labels                map[string]string
	Parameters            map[string]string
	IsIncremental         bool
	CompressionMethod     string
	NewlyUploadedDataSize int64 `json:",string"`
	ReUploadedDataSize    int64 `json:",string"`

	ProcessingBlocks *ProcessingBlocks

	Blocks     []BlockMapping `json:",omitempty"`
	SingleFile BackupFile     `json:",omitempty"`
}
```

the struct here will be used to Unmarshal the json string when getting config from the output
```
// BackupVolumeGet inspects a backup volume config with the given volume config URL
func (btc *BackupTargetClient) BackupVolumeGet(backupVolumeURL string, credential map[string]string) (*BackupVolume, error) {
	output, err := btc.ExecuteEngineBinary("backup", "inspect-volume", backupVolumeURL)
	if err != nil {
		if types.ErrorIsNotFound(err) {
			return nil, nil
		}
		return nil, errors.Wrapf(err, "error getting backup volume config %s", backupVolumeURL)
	}
	return parseBackupVolumeConfig(output)
}

// parseBackupConfig parses a backup config
func parseBackupConfig(output string) (*Backup, error) {
	backup := new(Backup)
	if err := json.Unmarshal([]byte(output), backup); err != nil {
		return nil, errors.Wrapf(err, "error parsing one backup config: \n%s", output)
	}
	return backup, nil
}
```

So I just follow the same type like `Size` and `VolumeSize`
```
type Backup struct {
...
	Size                   string               `json:"size"`
...
	VolumeName             string               `json:"volumeName"`
	VolumeSize             string               `json:"volumeSize"`
...
	NewlyUploadedDataSize  string               `json:"newlyUploadedDataSize"`
	ReUploadedDataSize     string               `json:"reUploadedDataSize"`
}
```

Reply:
OK! I see. Let's keep it as it is.